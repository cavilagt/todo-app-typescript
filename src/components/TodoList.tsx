import React from "react";

import Todo from "./Todo";
import { TodoType } from "../types/State";

type TodoListProps = {
  todos?: TodoType[];
}

const TodoList:React.FunctionComponent<TodoListProps> = ({todos =[] }) => {
  return (
    <div className="ui segments">
      {
        todos.length === 0
        ? <div className="ui segment">There are no more tasks</div>
        : todos.map( todo => (
          <Todo key={todo.uid} {...todo} />))
      }
  </div>
  );
};

export default TodoList;
