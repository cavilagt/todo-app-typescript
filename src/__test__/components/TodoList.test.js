import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import TodoList from "../../components/TodoList";

describe("<TodoList /> component", ()=> {
  
  afterEach(cleanup);

  it("reders without crashing",()=>{
    const roodDiv = document.createElement("div");
    ReactDOM.render( <TodoList />, roodDiv );
  });

  it("renders correctly empty", ()=>{
    const roodDiv = document.createElement("div");
    const { container } = render( <TodoList />, roodDiv );

    expect( container ).toMatchSnapshot();
  });

  it("renders correctly with todos", ()=>{
    const roodDiv = document.createElement("div");
    const todo = [{title:'Title todo test', status:"completed", uid:"1234", description:"A description, just for testing"}];
    const { container } = render( <TodoList {...todo} />, roodDiv );

    expect( container ).toMatchSnapshot();
  });

});