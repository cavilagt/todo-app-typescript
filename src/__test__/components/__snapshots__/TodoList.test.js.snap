// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> component renders correctly empty 1`] = `
<div>
  <div
    class="ui segments"
  >
    <div
      class="ui segment"
    >
      There are no more tasks
    </div>
  </div>
</div>
`;

exports[`<TodoList /> component renders correctly with todos 1`] = `
<div>
  <div
    class="ui segments"
  >
    <div
      class="ui segment"
    >
      There are no more tasks
    </div>
  </div>
</div>
`;
